File: db4oj/core/src/com/db4o/MWriteUpdateDeleteMembers.java
/* Copyright (C) 2004   db4objects Inc.   http://www.db4o.com */

package com.db4o;

import com.db4o.foundation.network.YapSocket;

final class MWriteUpdateDeleteMembers extends MsgD {
	public MWriteUpdateDeleteMembers() {
		super();
	}

	public MWriteUpdateDeleteMembers(MsgCloneMarker marker) {
		super(marker);
	}

	final boolean processMessageAtServer(YapSocket sock) {
	    YapStream stream = getStream();
		synchronized (stream.i_lock) {
			this.getTransaction().writeUpdateDeleteMembers(
			    readInt(),
				stream.getYapClass(readInt()),
				readInt(),
				readInt()
                );
		}
		return true;
	}
	
	public Object shallowClone() {
		return shallowCloneInternal(new MWriteUpdateDeleteMembers(MsgCloneMarker.INSTANCE));
	}
}

File: db4on/core/src/core/com/db4o/cs/messages/MWriteUpdateDeleteMembers.cs
namespace com.db4o.cs.messages
{
	public sealed class MWriteUpdateDeleteMembers : com.db4o.cs.messages.MsgD
	{
		public sealed override bool ProcessMessageAtServer(com.db4o.foundation.network.YapSocket
			 sock)
		{
			com.db4o.YapStream stream = GetStream();
			lock (stream.i_lock)
			{
				this.GetTransaction().WriteUpdateDeleteMembers(ReadInt(), stream.GetYapClass(ReadInt
					()), ReadInt(), ReadInt());
			}
			return true;
		}
	}
}
